code/modules/mob/living/carbon/human/diona_gestalt.dm

```diff
@@ -210,6 +207,81 @@
 		verbs.Remove(/mob/living/carbon/human/proc/gestalt_set_name)
 
 
+/mob/living/carbon/human/proc/diona_detach_nymph()
+	set name = "Detach nymph"
+	set desc = "Allows you to detach specific nymph, and control it."
+	set category = "Abilities"
+
+	if(nutrition <= 150)
+		to_chat(src, span("warning", "You lack nutrition to perform this action!"))
+		return
+	if(DS.stored_energy <= 60)
+		to_chat(src, span("warning", "You lack energy to perform this action!"))
+		return
+	// Choose our limb to detach
+	var/list/exclude = organs_by_name - list("groin", "chest", "l_arm", "r_arm", "l_leg", "r_leg")
+	var/choice =  input(src, "Choose a limb to detach?", "Limb detach") as null|anything in exclude
+	if(!choice)
+		return
+	var/obj/item/organ/external/O = organs_by_name[choice]
+	if(!O || O.is_stump())
+		to_chat(src, span("warning", "Cannot detach that!"))
+		return
+	
+	// Get rid of our limb and replace with stump
+	var/obj/item/organ/external/stump/stump = new (src, 0, O)
+	O.removed(null, TRUE)
+	organs |= stump
+	stump.update_damages()
+	O.post_droplimb(src)
+	// If we got parent organ - drop it too
+	if(O.parent_organ && O.parent_organ != "chest")
+		var/obj/item/organ/external/parent = organs_by_name[O.parent_organ]
+		var/obj/item/organ/external/stump/parent_stump = new (src, 0, parent)
+		parent.removed(null, TRUE)
+		organs |= parent_stump
+		parent_stump.update_damages()
+		parent.post_droplimb(src)
+		qdel(parent)
+	to_chat(src, span("notice", "You detach [O.name] nymph from your body."))
+	qdel(O)
+
+	// Spawn new nymph 
+	var/mob/living/carbon/alien/diona/M = locate(/mob/living/carbon/alien/diona) in contents
```

Comment:
this isn't spawning and it can technically return null.

Reply:
Shouldn't happen, as in that case we won't have organs to detach. But I will add a check